This patch fixes undefined ksh behaviour -> test with >= 4 arguments.
( http://www.opengroup.org/onlinepubs/009695399/utilities/test.html )

catched error: ./mkbuiltins[122]: [: #: unexpected operator/operand

--- klibc-1.1.1/ash/mkbuiltins.orig	2005-09-06 22:49:34.000000000 +0200
+++ klibc-1.1.1/ash/mkbuiltins	2005-09-30 21:41:07.909992750 +0200
@@ -108,7 +108,7 @@
 	}
 	echo 'int '"$func"'(int, char **);' >&4
 	while
-		[ $# != 0 -a "$1" != '#' ]
+		[ $# != 0 ] && [ "$1" != '#' ]
 	do
 		[ "$1" = '-s' ] && {
 			specials="$specials $2 $func"
